torch/testing/_internal/common_nn.py

```diff
@@ -4566,7 +4622,8 @@ def check_jacobian(self, module, input, jacobian_input=True):
         numerical_t = list(iter_tensors(numerical))
 
         # TODO: compare structure
-        if input.numel() != 0:
+        input_tuple = input if isinstance(input, tuple) else (input,)
+        if any(t.numel() != 0 for t in input_tuple):
```

Comment:
It looks like the `input` to `check_jacobian` is *always* the tupled inputs. i.e., on line 4953 we have `test_case.check_jacobian(module, input_tuple, self.jacobian_input)`, which looks like the only callsite of check_jacobian.

Is that true? If so, we should update the `check_jacobian` function to take in `self, module, inputs_tuple` and remove this conditional.

Reply:
This is something I am not sure about. Before this change, check_jacobian accept both individual tensor and tuple of tensors. After this change, we no longer have individual tensor being passed in, but should we remove the support of individual tensor for check_jacobian? I prefer keep the support rather than removing, what do you think?

Reply:
I find it confusing when `input` can be "either a Tensor or a Tuple of Tensors", especially if our testing code never actually exercises the "Tensor" path. I'm fine if you'd like to leave it in -- in that case, would prefer some type annotations so that a code reader has some idea of what is going on: https://github.com/pytorch/pytorch/pull/44853#discussion_r492341897

Reply:
I will add type-annotate. I gained a lot of benefit from downstream functions (eg. to_gpu())  which supported tuple tensor already, that made my change much easier. So I think it would be better to keep what we already supported.